modules/extensions-support/src/main/java/org/mule/module/extension/internal/metadata/DefaultMetadataContext.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.module.extension.internal.metadata;
+
+import org.mule.api.connection.ConnectionException;
+import org.mule.api.connector.ConnectionManager;
+import org.mule.api.metadata.MetadataContext;
+import org.mule.extension.api.runtime.ConfigurationInstance;
+
+import java.util.Optional;
+
+/**
+ * Default immutable implementation of {@link MetadataContext}
+ *
+ * @since 1.0
+ */
+public class DefaultMetadataContext implements MetadataContext
```

Comment:
I don't know if is the correct name, although we are using it for Metadata fetching, what it does is not directly linked to metadata fetching, and maybe this can be reused in another scenario where a config or connection is needed to be injected.